gateway/server.go

```diff
@@ -207,7 +207,22 @@ func main() {
 	}
 
 	metricsHandler := metrics.PrometheusHandler()
-	r.Handle("/metrics", metricsHandler)
+
+	//Create a new server to serve /metrics endpoint
+	go func() {
+		routes := mux.NewRouter()
+		routes.Handle("/metrics", metricsHandler)
+		port := 8082
+		s := &http.Server{
+			Addr:           fmt.Sprintf(":%d", port),
+			ReadTimeout:    config.ReadTimeout,
+			WriteTimeout:   config.WriteTimeout,
+			MaxHeaderBytes: http.DefaultMaxHeaderBytes,
+			Handler:        routes,
+		}
```

Comment:
How do you plan on stopping this goroutine when the main thread stops? The server should be defined outside of the goroutine, there should be some kind of signal catch, and then sending `s.Shutdown()` to stop this gracefully.

Reply:
I think when main thread stops it will also stop the go routines launched from main thread, it may not do the cleanup on the resources used.  I will implement the server to shutdown gracefully.

Reply:
I think we could set a shorter, static timeout value for the HTTP server given that this is not serving function requests.

Reply:
I have update the server to use `30s` for `readTimeout` and `writeTimeout`

Reply:
`30s` seems too high for this purpose, let's reduce it to `5s`?